go-controller/pkg/ovn/gateway_init.go

```diff
@@ -155,9 +158,34 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 		// router: UDP, TCP, SCTP
 		for _, proto := range enabledProtos {
 			if gatewayProtoLBMap[proto] == "" {
-				gatewayProtoLBMap[proto], stderr, err = util.RunOVNNbctl("--", "create",
+				if config.Gateway.Mode != config.GatewayModeLocal {
+					gatewayProtoLBMap[proto], stderr, err = util.RunOVNNbctl("--", "create",
+						"load_balancer",
+						fmt.Sprintf("external_ids:%s_lb_gateway_router=%s", proto, gatewayRouter),
+						fmt.Sprintf("protocol=%s", strings.ToLower(string(proto))),
+						"options:lb_force_snat_ip=router_ip")
+					if err != nil {
+						return fmt.Errorf("failed to create load balancer for gateway router %s for protocol %s: "+
+							"stderr: %q, error: %v", gatewayRouter, proto, stderr, err)
+					}
+				} else {
+					klog.V(5).Infof("Gatway is local making single set of GW lbs")
+					gatewayProtoLBMap[proto], stderr, err = util.RunOVNNbctl("--", "create",
+						"load_balancer",
+						fmt.Sprintf("external_ids:%s_lb_gateway_router=%s", proto, gatewayRouter),
+						fmt.Sprintf("protocol=%s", strings.ToLower(string(proto))))
+					if err != nil {
+						return fmt.Errorf("failed to create load balancer for gateway router %s for protocol %s: "+
+							"stderr: %q, error: %v", gatewayRouter, proto, stderr, err)
+					}
+				}
+			}
+			// Make Local LBs if needed
+			if gatewayProtoLocalLBMap[proto] == "" && config.Gateway.Mode != config.GatewayModeLocal {
+				// Create LB's for externalTrafficPolicy=local mode no snat
+				gatewayProtoLocalLBMap[proto], stderr, err = util.RunOVNNbctl("--", "create",
 					"load_balancer",
-					fmt.Sprintf("external_ids:%s_lb_gateway_router=%s", proto, gatewayRouter),
+					fmt.Sprintf("external_ids:%s_lb_gateway_router=%s_local", proto, gatewayRouter),
```

Comment:
Instead of hardcoding _local I'd use a function here, something like localLBForGWRouter that returns the name, and use it when you look for the lb